pkg/apis/application/v1alpha1/types.go

```diff
@@ -3130,6 +3133,16 @@ func (w *SyncWindow) Validate() error {
 	return nil
 }
 
+func (w *SyncWindow) Hash() uint64 {
+	var windowBuffer bytes.Buffer
+	enc := gob.NewEncoder(&windowBuffer)
+	err := enc.Encode(w)
+	if err != nil {
+		return 0
```

Comment:
should return `err`. If there is a problem with this code, the hash will be 0 for 2 windows leading to a `window '%s':'%s':'%s' already exists, update or edit` error. This will not be easy to figure out why.

and it requires a test for error handling